sdk/python/feast/infra/registry/caching_registry.py

```diff
@@ -425,12 +426,25 @@ def list_projects(
         return self._list_projects(tags)
 
     def refresh(self, project: Optional[str] = None):
-        self.cached_registry_proto = self.proto()
-        self.cached_registry_proto_created = _utc_now()
+        if self._refreshing:  # Skipping refresh if already in progress
```

Comment:
I think it is better to log here. it will be helpful to debug or observe the pattern.

Reply:
Done!